3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +23,70 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`HTTP Error: ${response.status}`);
+    }
+    return await response.json();
+  } catch (error) {
+    console.log(error.message);
+    throw error;
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchAndPopulatePokemons() {
+  const getPokemonButton = document.createElement('button');
+  getPokemonButton.type = 'button';
+  getPokemonButton.textContent = 'Get Pokemon!';
+  getPokemonButton.classList.add('get-pokemon-button');
+  document.body.appendChild(getPokemonButton);
+
+  const selectPokemon = document.createElement('select');
+  selectPokemon.classList.add('select-pokemon');
+  document.body.appendChild(selectPokemon);
+
+  let data = null;
+  const urlApi = 'https://pokeapi.co/api/v2/pokemon?limit=151';
+  getPokemonButton.addEventListener('click', async () => {
+    try {
+      data = await fetchData(urlApi);
+    } catch (error) {
+      console.error(error.message);
+    }
+    const results = data.results;
+    for (const pokemon of results) {
+      const option = document.createElement('option');
+      option.textContent = pokemon.name;
+      option.value = results.indexOf(pokemon) + 1;
+      selectPokemon.appendChild(option);
+    }
+  });
+  selectPokemon.addEventListener('change', fetchImage);
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchImage(image) {
+  const previousPokemon = document.querySelector('img');
+  if (previousPokemon) {
+    document.body.removeChild(previousPokemon);
+  }
+  const currentPokemon = `${image.currentTarget.value}`;
```

Comment:
This could be the url if you followed my earlier comment (and you don't need a template literal here: there is interpolation (i.e. `${...}`) used).

```suggestion
  const url= event.currentTarget.value;
```